src/com/seafile/seadroid2/cameraupload/CloudLibraryAdapter.java

```diff
@@ -0,0 +1,118 @@
+package com.seafile.seadroid2.cameraupload;
+
+import android.view.View;
+import android.widget.ImageView;
+import com.seafile.seadroid2.R;
+import com.seafile.seadroid2.data.SeafRepo;
+import com.seafile.seadroid2.ui.adapter.ReposAdapter;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+/**
+ * Cloud library adapter
+ */
+public class CloudLibraryAdapter extends ReposAdapter {
+
+    protected LinkedHashMap<SeafRepo, Boolean> repos = new LinkedHashMap<SeafRepo, Boolean>();
```

Comment:
This map is a `repo -> is_repo_selected` map, but when choosing the target repo for camera upload, there would only be one library that would be selected, so it doesn't make sense to use a hash map here.